### PR TITLE
vkd3d: Require and use VK_KHR_copy_commands2

### PR DESCRIPTION
**Pull request by hakzsam**
Based on https://github.com/HansKristian-Work/vkd3d-proton/pull/958

### CODE REVIEW

File: libs/vkd3d/vulkan_procs.h [hunk 1]
```diff
@@ -225,6 +225,10 @@ VK_DEVICE_EXT_PFN(vkCmdEndRenderPass2KHR)
 VK_DEVICE_EXT_PFN(vkCmdNextSubpass2KHR)
 VK_DEVICE_EXT_PFN(vkCreateRenderPass2KHR)
 
+/* VK_KHR_bind_memory2 */
+VK_DEVICE_EXT_PFN(vkBindBufferMemory2KHR);
```
Comment (HansKristian-Work) [inline libs/vkd3d/vulkan_procs.h:229]: In the following commit, the legacy entry points should be removed from vulkan_procs.h to ensure that we don't accidentally use the legacy entry points, and verifies that we are not still using them.

File: libs/vkd3d/vulkan_procs.h [hunk 2]
```diff
@@ -229,6 +229,14 @@ VK_DEVICE_EXT_PFN(vkCreateRenderPass2KHR)
 VK_DEVICE_EXT_PFN(vkBindBufferMemory2KHR);
 VK_DEVICE_EXT_PFN(vkBindImageMemory2KHR);
 
+/* VK_KHR_copy_commands2 */
+VK_DEVICE_EXT_PFN(vkCmdBlitImage2KHR);
```
Comment (HansKristian-Work) [inline libs/vkd3d/vulkan_procs.h:233]: Similar concern here w.r.t. entry points.

File: libs/vkd3d/command.c [hunk 3]
```diff
@@ -6177,6 +6212,9 @@ static void STDMETHODCALLTYPE d3d12_command_list_CopyTextureRegion(d3d12_command
         dst_format = dst_resource->format;
         src_format = src_resource->format;
 
+        image_copy.sType = VK_STRUCTURE_TYPE_IMAGE_COPY_2_KHR;
```
Comment (HansKristian-Work) [inline libs/vkd3d/command.c:6215]: This looks a little goofy. Shouldn't vk_image_copy_from_d3d12 fill this in?

File: libs/vkd3d/command.c [hunk 4]
```diff
@@ -8695,8 +8790,9 @@ static void STDMETHODCALLTYPE d3d12_command_list_SetPredication(d3d12_command_li
     VkPipelineStageFlags dst_stages, src_stages;
     struct vkd3d_scratch_allocation scratch;
     VkAccessFlags dst_access, src_access;
+    VkCopyBufferInfo2KHR copy_info;
     VkMemoryBarrier vk_barrier;
-    VkBufferCopy copy_region;
+    VkBufferCopy2KHR copy_region;
```
Comment (HansKristian-Work) [inline libs/vkd3d/command.c:8795]: Inverse christmas tree ordering.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Comments.

File: libs/vkd3d/vulkan_procs.h (hunk 1 above)
Comment (hakzsam) [reply libs/vkd3d/vulkan_procs.h:229]: I can do that but it seems like some legacy entrypoints stil remain like vkCreateRenderPass().

File: libs/vkd3d/vulkan_procs.h (hunk 2 above)
Comment (hakzsam) [reply libs/vkd3d/vulkan_procs.h:225]: Removed.

File: libs/vkd3d/command.c (hunk 3 above)
Comment (hakzsam) [reply libs/vkd3d/command.c]: Fixed.

File: libs/vkd3d/command.c (hunk 4 above)
Comment (hakzsam) [reply libs/vkd3d/command.c:8790]: Fixed.

**Review by HansKristian-Work — APPROVED** [review]
LGTM.

**Pull request merged**